raiden/tests/integration/network/proxies/test_payment_channel.py

```diff
@@ -231,8 +229,8 @@ def test_payment_channel_outdated_channel_close(  # pylint: disable=unused-argum
     # check the settlement timeouts again
     assert channel_proxy_1.settle_timeout() == TEST_SETTLE_TIMEOUT_MIN
 
-    # update transfer
-    chain.wait_until_block(target_block_number=client.block_number() + TEST_SETTLE_TIMEOUT_MIN)
+    # update transfer -- we need to wait on +1 since we use the latest block on parity
```

Comment:
This comment is a bit vague... we provide the `target_block_number`... whats the relation betwen +1 & latest and the block number we're passing?

Reply:
If you don't get it then this means I should clarify further. Does the part below make sense?

So in at the moment whenever we check preconditions, we check them on a different block for parity and a different one for geth. It's the so called "checking_block" seen [here](https://github.com/raiden-network/raiden/blob/98ae425a3f39c70d1a6272b8da9bca39bbda3b73/raiden/network/proxies/token_network.py#L223).

The checking block is defined below:

https://github.com/raiden-network/raiden/blob/98ae425a3f39c70d1a6272b8da9bca39bbda3b73/raiden/network/rpc/client.py#L839-L849

It's always `pending` in geth (this is how estimate gas works there) but `latest` in parity as in parity there can exist no pending block. Before you ask we can't configure geth to also be on latest. They don't follow the spec and don't allow a block identifier argument for estimate gas. Issue [here](https://github.com/ethereum/go-ethereum/issues/2586).

The code before was waiting for the settlement block. And once it arrived it started the settle() call, which contains a gas estimation. But `estimate_gas` for settle on the latest block being the settle block number will fail (see the [contract code](https://github.com/raiden-network/raiden-contracts/blob/03024ac310f34b9d8f5ddd064413684416223845/raiden_contracts/data/source/raiden/TokenNetwork.sol#L697)). We did not see it in geth since it's estmating on the pending block (which is latest +1).


Now I gotta find a way to explain this in a shorter way in the comment ... :sweat_smile:

Reply:
Understood... the difference between clients is horrible!

Reply:
I think you can just link your comment in a comment :laughing: should be sufficient